src/cbmc/cbmc_parse_options.cpp

```diff
@@ -545,7 +547,7 @@ int cbmc_parse_optionst::doit()
   }
 
   int get_goto_program_ret =
-    get_goto_program(goto_model, options, cmdline, log, ui_message_handler);
+    get_goto_program(goto_model, options, cmdline, log, message_handler);
```

Comment:
Are we actually passing a `messaget` and a `message_handlert` here?!

Reply:
the function needs a `messaget` and a `ui_message_handlert`, so `log.get_message_handler()` wouldn't have the right type.

Reply:
This isn't at all a questionable piece of code you're introducing here, but I'm really wondering why we have this interface. It should be sufficient to pass a `message_handlert` or `ui_message_handlert` and then just create a `messaget` locally in that function.

Reply:
`get_goto_program` is a non-static member, so neither log nor message_handler should be passed.

Reply:
`get_goto_program` is static in `cbmc_parse_optionst`. I will add a commit to construct the messaget from the message handler instead of passing both.